### DISCUSSION TITLE
Zxing.net generate qr code xamarin.android null reference

### DISCUSSION THREAD

**Discussion by youssefyaza** [micjahn/ZXing.Net#393] [Q&A] [answered]
I have a problem generating qr code. I used a sample from site: 
https://www.c-sharpcorner.com/article/generate-barcode-and-qr-code-in-xamarin-android/

I have used the exact same code as in the example but I got a null reference error at this line:
`bitmapMatrix = new MultiFormatWriter().encode(message, BarcodeFormat.QR_CODE, size, size);`

The problem only occurs BarcodeFormat.QR_CODE the others work well. 

In my project I have installed the next packages (the newest versions):
- Zxing.net
- Xamarin.AndroidX.AppCompat
- Xamarin.Essentials
- Xamarin.Google.Android.Material

The null reference exception comes from the writer.encode message (please see my debug info):
![zxingwriter](https://user-images.githubusercontent.com/22446075/149917285-b8fe3877-edc6-4ffa-8c7f-b1e728cc6153.PNG)

Does anyone experienced this problem?

**Comment by micjahn** [comment, marked as answer]
https://stackoverflow.com/questions/70741623/zxing-net-generate-qr-code-xamarin-android-null-reference/70761473#70761473
It's a bug inside the static initialization routines which occurs if some encodings (character sets) are missing. The next version (>0.16.7) will fix it.
